sympy/series/limits.py

```diff
@@ -136,6 +136,11 @@ def doit(self, **hints):
         if not e.has(z):
             return e
 
+        # limit can work with binomial by converting to factorial first
+        if (e.has(binomial) and not (hints.get('sequence', True)
```

Comment:
I tried something similar. As you said, returning in ``factorial`` form did not help, so I tried it with ``gamma``, which appears to give wrong result. You do need not to call it anywhere. It is already called [here](https://github.com/sympy/sympy/tree/master/sympy/series/gruntz.py#L411). Also, You need to implement ``_eval_rewrite_as_tractable`` and not ``eval_as_tractable``.